### PR TITLE
Feature/nick/categories

### PR DESCRIPTION
**Pull request by nschroder26**
Created categories model. Complete with standard CRUD methods, restricted to Admin roles only

### CODE REVIEW

**Comment by chrisconcepcion** [comment]
Need screenshots of the ui for categories, all view changes require screenshots.

File: app/views/categories/_create.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div class="card category-form">
```
Comment (chrisconcepcion) [inline app/views/categories/_create.html.erb:1]: I am not sure how you get to this action but usually what you do is you make a form partial and you use it for both new and edit to render the form. The idea behind doing this is called DRY(Do not repeat yourself). It's common practice, here's a not so helpful wiki about it https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

File: app/models/category.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class Category < ApplicationRecord
+end
```
Comment (chrisconcepcion) [inline app/models/category.rb:3]: We need validation for presence and uniqueness on name please. Can't have two categories of the same name and we cant have blank categories.

File: app/controllers/categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      redirect_to categories_path, notice: "#{@category.name} has been created"
+    else
+      render 'new'
```
Comment (chrisconcepcion) [inline app/controllers/categories_controller.rb:18]: There is no new partial

**Comment by nschroder26** [comment]
image
image
UI screenshots

File: app/views/categories/_create.html.erb (hunk 1 above)
Comment (nschroder26) [reply app/views/categories/_create.html.erb:1]: updated new/edit actions to use a shared partial

File: app/models/category.rb (hunk 2 above)
Comment (nschroder26) [reply app/models/category.rb:3]: added validation check for both presence and uniqueness for category names

File: app/controllers/categories_controller.rb (hunk 3 above)
Comment (nschroder26) [reply app/controllers/categories_controller.rb:18]: added new partial

**Pull request merged**